### PR TITLE
[path_parsing] deprecate utility functions that should be private

### PR DESCRIPTION
**Pull request by kevmoo**
Towards https://github.com/flutter/flutter/issues/157940

### CODE REVIEW

File: third_party/packages/path_parsing/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 1.0.4
+
+* Deprecate utility functions `blendPoints` and `reflectedPoint`.
```
Comment (stuartmorgan) [inline third_party/packages/path_parsing/CHANGELOG.md:3]: Deprecates

**Review by stuartmorgan — COMMENTED** [review]
There are other `library_private_types_in_public_api` violations in this file, including others using this type; it seems like it would make more sense to fix all of them at once rather than a series of deprecations.

File: third_party/packages/path_parsing/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+## 1.0.4
```
Comment (stuartmorgan) [inline third_party/packages/path_parsing/CHANGELOG.md:1]: Deprecation is defined as a minor version in semver.

**Comment by kevmoo** [comment]
Did see the other ones. Happy to fix, if there are straight forward

**Review by domesticmouse — APPROVED** [review]

**Comment by stuartmorgan** [comment]
test-exempt: code refactor with no semantic change

**Review by stuartmorgan — APPROVED** [review]
LGTM

**Pull request merged**